cash-register/js/CashRegister.js

```diff
@@ -0,0 +1,109 @@
+var cashRegister = {
+  total: 0,
+  lastItemCost: 0,
+  lastItemName: '',
+  lastItemQuantity: 0,
+  bill: {},
+
+  // add to cash register total, taking in name of item & quantity
+  add: function(itemName, quantity) {
+
+    // keep track of what the last item and quantity was for bill
+    this.lastItemName = itemName;
+    this.lastItemQuantity = quantity;
+
+    // looping through itemList to match name to find price & add to total
+    for (var item in itemList) {
+      if (itemName === itemList[item].name) {
+        this.lastItemCost = itemList[item].price * quantity;
+        this.total += this.lastItemCost;
+
+      }
+    }
+
+    // add item to bill
+
+    // check if item exists on bill
+
+    var billItemCheck = false;
+
+    for (var billItem in this.bill) {
```

Comment:
Okay, today's lesson - Leveraging other libraries (aka `Don't reinvent the wheel!`).
There's a very helpful utility library called `lodash` (https://lodash.com) which provides very useful functions across strings, arrays and objects. if you include that script in your HTML file then you could call the `_.includes()` function (defined at https://lodash.com/docs#includes) which would provide the same functionality as this `billItemCheck` loop for you. :)